frontend/expr_inferrer.py

```diff
@@ -124,6 +128,14 @@ def _infer_add(left_type, right_type, lineno, solver):
     result_type = solver.new_z3_const("addition_result")
     solver.add(axioms.add(left_type, right_type, result_type, solver.z3_types),
                fail_message="Addition in line {}".format(lineno))
+
+    # Add soft constraint in case of list addition
```

Comment:
But we can have soft constraints that make all the types (operands and result) equal to each other (hence not invoking subtype at all).